tools/pipelines/templates/download-pipeline-artifact.yml

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+# Licensed under the MIT License.
+
+# download-pipeline-artifact pipeline
```

Comment:
Nit: since this is specifically downloading api-extractor artifacts, we should probably make the template name a bit more explicit to make that clear. Alternatively, this template could take an `artifactName` parameter and be more general. Since we're hard-coding `allowPartiallySucceededBuilds` to true for reasons specific to our api-extractor artifact requirements, I would probably lean towards the first option.